app/app.go

```diff
@@ -1577,30 +1577,47 @@ func (app *App) SetEnvs(setEnvs bind.SetEnvArgs) error {
 	if len(setEnvs.Envs) == 0 {
 		return nil
 	}
+
 	for _, env := range setEnvs.Envs {
 		err := validateEnv(env.Name)
 		if err != nil {
 			return err
 		}
 	}
+
 	if setEnvs.Writer != nil {
 		fmt.Fprintf(setEnvs.Writer, "---- Setting %d new environment variables ----\n", len(setEnvs.Envs))
 	}
+
+	if setEnvs.PruneUnused {
+		for name, value := range app.Env {
+			ok := envInSet(name, setEnvs.Envs)
+			// only prune variables managed by requested
+			if !ok && value.ManagedBy == setEnvs.ManagedBy {
+				delete(app.Env, name)
```

Comment:
I suggest to we have a message in streaming to inform that some environment variable will be deleted. @pedrokiefer